### ISSUE TITLE
Observed data / Reload under same settings: inconsistent state

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/OSPSuite.Core#1068]
[Organs1.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/6569866/Organs1.xlsx) contains observed data for **Brain** and **Liver**.
[Organs2.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/6569867/Organs2.xlsx) contains observed data for **Brain** and **Heart**.
I import Organs1.xlsx first.
![grafik](https://user-images.githubusercontent.com/25061876/120185959-25d7d100-c213-11eb-981b-afb371d529a3.png)
Now I call "Reload under the same settings" and select Organs2.xlsx
The confirmation dialog is shown
![grafik](https://user-images.githubusercontent.com/25061876/120186128-61729b00-c213-11eb-8f68-2f037bbc8033.png)
I click OK. Another confirmation dialog is shown
![grafik](https://user-images.githubusercontent.com/25061876/120186296-9aab0b00-c213-11eb-9d11-be180fb6516a.png)
Here, I can see that the new Observed data set for heart was already added bevor I confirm the deletion of Liver data.
Now I select **No** - and end up in inconsistent state, where observed data from both Organs1.xlsx and Organs2.xlsx is in the project.
![grafik](https://user-images.githubusercontent.com/25061876/120186452-cfb75d80-c213-11eb-853f-e3ad043951bf.png)

So I think the update order is wrong: new datasets are added first (and changed datasets are probably updated as well) and only after that datasets which should be deleted are deleted.
It should be done the other way around: try to delete first and only if the user confirms deletion - continue with adding/updating

**Comment by Yuri05** [comment]
It becomes even more interesting if due to the data update more than 1 observed data set should be deleted from the project. Than the prompt "Delete observed data set XY" comes for every single data set to be deleted. 
That's not good. "Reload under the same settings" should be like a database transaction: either completely performed or completely cancelled.
Actually I think that the second "really delete?" prompt is already an overkill. The "Reload data" overview already shows that some datasets will be deleted.
![grafik](https://user-images.githubusercontent.com/25061876/120187646-5caee680-c215-11eb-9ee6-7f33eda8c8da.png)
But if we really think that we need the second confirmation: it should be shown only once for all data sets to be deleted.

**Comment by georgeDaskalakis** [comment]
Add a warning in case at least one dataset is to be deleted, between the question and the table of datasets. Then delete what is to be deleted without asking again.

**Comment by PavelBal** [comment]
> The only problem we could have is that the user clicked OK when he/she really wanted to click Cancel. 

Yes, well, not our problem if the user clicks not what he actually wants to click :D

Otherwise - I agree, no need to show the second dialogue.
